File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int n = A.length;
        int[] freq = new int[n+1];
        int[] ans = new int[n];
        int temp = 0;
        for(int i=0;i<n;i++){
            freq[A[i]-1]++;
            if(freq[A[i]-1] == 2){
                temp++;
            }
            freq[B[i]-1]++;
            if(freq[B[i]-1] == 2){
                temp++;
            }
            ans[i] = temp;
        }
        return ans;
    }
}

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        n = len(A)
        freq = [0]*(n+1)
        ans = [0]*(n)
        temp = 0
        for i in range(n):
            freq[A[i]-1]+=1
            if freq[A[i]-1] == 2:
                temp+=1
            freq[B[i]-1]+=1
            if freq[B[i]-1] == 2:
                temp+=1
            ans[i] = temp
        return ans